### PR TITLE
[I18N] Add permissions

### PR DESCRIPTION
**Pull request by mfrachet**
### What does it do?

Conditionally shows content based on user permissions.

### Why is it needed?

Being sure that not everybody can do specific actions

### CODE REVIEW

File: packages/strapi-plugin-i18n/admin/src/permissions.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+const i18nPermissions = {
```
Comment (mfrachet) [inline packages/strapi-plugin-i18n/admin/src/permissions.js:1]: ❓ I'm not sure about the shape of this object. It looks to resolve correctly the permissions however 🤷🏻‍♂️

File: packages/strapi-plugin-i18n/admin/src/containers/SettingsPage/index.js [hunk 2]
```diff
@@ -1,136 +1,35 @@
...
+
+  if (isLoading) {
+    return (
+      <div>
+        <p>{formatMessage({ id: getTrad('Settings.permissions.loading') })}</p>
+      </div>
+    );
+  }
+
+  if (!canRead) {
```
Comment (mfrachet) [inline packages/strapi-plugin-i18n/admin/src/containers/SettingsPage/index.js:23]: When the user is not allowed to read the content, I'm displaying a message. Is this something we want or should I just redirect the user?

File: packages/strapi-plugin-i18n/admin/src/containers/SettingsPage/LocaleSettingsPage.js [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+const LocaleSettingsPage = ({ canCreate, canDelete, canUpdate }) => {
+  const {
+    isDeleting,
+    isDeleteModalOpen,
+    deleteLocale,
+    showDeleteModal,
+    hideDeleteModal,
+  } = useDeleteLocale();
+
+  const { isEditing, isEditModalOpen, editLocale, showEditModal, hideEditModal } = useEditLocale();
```
Comment (mfrachet) [inline packages/strapi-plugin-i18n/admin/src/containers/SettingsPage/LocaleSettingsPage.js:24]: ℹ️ If LOCs start to become a problem, we can eventually split this hook in two: one for dealing with the modal, and one for the edit code. Then, we could only deal with the modal one in this file and move the one about the edition directly inside the `ModalEdit`. Same applies for the delete one

File: packages/strapi-plugin-i18n/admin/src/permissions.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+  create: [{ action: 'plugins::i18n.locale.create', subject: null }],
+  delete: [{ action: 'plugins::i18n.locale.delete', subject: null }],
+  update: [{ action: 'plugins::i18n.locale.update', subject: null }],
+  read: [{ action: 'plugins::i18n.locale.refead', subject: null }],
+};
```
Comment (soupette) [inline packages/strapi-plugin-i18n/admin/src/permissions.js:6]: Can you add another key to permissions `accessMain: [{ action: 'plugins::i18n.locale.read', subject: null}, { action: 'plugins::i18n.locale.create, subject: null}]` and the add the following to the index.js:

```
settings: {
  global: {
    links: [
      {
        title: {
          id: getTrad('plugin.name'),
          defaultMessage: 'Internationalization',
        },
        name: 'internationalization',
        to: `${strapi.settingsBaseURL}/internationalization`,
        Component: () => <SettingsPage />,
        permissions: pluginPermissions.accessMain,
      },
    ],
  },
```

By doing so, if a user does not have the permission to either create or read the settings the link won't be displayed in the settings menu.

**Review by soupette — CHANGES_REQUESTED** [review]

File: packages/strapi-plugin-i18n/admin/src/containers/SettingsPage/LocaleSettingsPage.js (hunk 3 above)
Comment (soupette) [reply packages/strapi-plugin-i18n/admin/src/containers/SettingsPage/LocaleSettingsPage.js:24]: I'll let you decide what you think is best.

File: packages/strapi-plugin-i18n/admin/src/containers/SettingsPage/index.js (hunk 2 above)
Comment (soupette) [reply packages/strapi-plugin-i18n/admin/src/containers/SettingsPage/index.js:23]: Usually, we simply don't display the component.

**Review by soupette — APPROVED** [review]
LGTM

**Pull request merged**